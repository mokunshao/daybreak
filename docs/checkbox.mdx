---
title: Checkbox
---

import {useState} from 'react'
import { Playground, Props } from 'docz'
import Checkbox from '../src/checkbox/checkbox.tsx'

# Checkbox

## Standalone

<Playground>
  {()=>{
    const [checked1, setChecked1] = useState(true);
    return (
      <div>
        <Checkbox checked={checked1} onChange={(e) => setChecked1(e.target.checked)} />
      </div>
    );
  }}
</Playground>

## With children

<Playground>
  {()=>{
    const [checked2, setChecked2] = useState(false);
    return (
      <div>
        <Checkbox checked={checked2} onChange={(e) => setChecked2(e.target.checked)}>ü§îÔ∏èAgree?</Checkbox>
      </div>
    );
  }}
</Playground>

## Props

<Props of={Checkbox} />
