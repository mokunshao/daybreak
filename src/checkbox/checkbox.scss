@import '../common.scss';

.daybreak-checkbox {
  cursor: pointer;
  display: inline-flex;
  align-items: center;

  &-input {
    opacity: 0;
    height: 0;
    width: 0;
  }

  &-checkmark {
    display: inline-flex;
    height: 1em;
    width: 1em;
    margin-right: 0.2em;
    margin-left: 0.2em;
    border: 2px solid $black;
    transition: background-color 200ms;
  }

  &:hover &-checkmark {
    background-color: $grey;
  }

  &-input:checked ~ &-checkmark {
    background-color: $black;
    position: relative;

    &::after {
      content: '';
      position: absolute;
      top: 40%;
      left: 50%;
      width: 0.4em;
      height: 0.7em;
      border: solid white;
      border-width: 0 0.1em 0.1em 0;
      transform: translate(-50%, -50%) rotate(45deg);
    }
  }

  &-children {
    display: inline-flex;
  }
}
