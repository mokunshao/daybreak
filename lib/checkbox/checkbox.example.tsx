import React, { useState } from 'react';
import { Checkbox } from './checkbox';

export const CheckboxExample: React.FC = () => {
  const [checked1, setChecked1] = useState(true);
  const [checked2, setChecked2] = useState(false);
  const [checked3, setChecked3] = useState(false);
  return (
    <div>
      <Checkbox checked={checked1} onChange={(e) => setChecked1(e.target.checked)} />
      <br />
      <Checkbox checked={checked2} onChange={(e) => setChecked2(e.target.checked)} />
      <br />
      <Checkbox checked={checked3} onChange={(e) => setChecked3(e.target.checked)}>Agree?</Checkbox>
    </div>
  );
};

export default CheckboxExample;
