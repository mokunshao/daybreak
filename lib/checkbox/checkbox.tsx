import React, { InputHTMLAttributes } from 'react';
import './checkbox.scss';
import { classes } from '../utils/classes';
import { joinedClass } from '../utils/joinedClass';

const baseClass = joinedClass('checkbox');

interface Props extends InputHTMLAttributes<HTMLInputElement> {
  checked: boolean;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

export const Checkbox: React.FC<Props> = (props) => {
  const {
    checked, onChange, children, className, ...rest
  } = props;
  return (
    <label className={classes(baseClass(), className)}>
      <input className={baseClass('input')} type="checkbox" checked={checked} onChange={(e) => onChange(e)} {...rest} />
      <span className={baseClass('checkmark')} />
      {children && <div className={baseClass('children')}>{children}</div>}
    </label>
  );
};

export default Checkbox;
