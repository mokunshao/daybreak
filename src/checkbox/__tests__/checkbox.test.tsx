import React, { useState } from 'react';
import { render } from '@testing-library/react';
import { Checkbox } from '../checkbox';

function Example() {
  const [checked1, setChecked1] = useState(true);
  return (
    <div>
      <Checkbox checked={checked1} onChange={(e) => setChecked1(e.target.checked)} />
    </div>
  );
}

describe('Tabs', () => {
  it('exits', () => {
    expect(Checkbox).toBeTruthy();
    expect(<Example />).toBeTruthy();
  });

  it('match snapshot', () => {
    const { asFragment } = render(<Example />);
    expect(asFragment()).toMatchSnapshot();
  });
});
