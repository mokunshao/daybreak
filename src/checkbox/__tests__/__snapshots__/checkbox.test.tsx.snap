// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs match snapshot 1`] = `
<DocumentFragment>
  <div>
    <label
      class="daybreak-checkbox"
    >
      <input
        checked=""
        class="daybreak-checkbox-input"
        type="checkbox"
      />
      <span
        class="daybreak-checkbox-checkmark"
      />
    </label>
  </div>
</DocumentFragment>
`;
